nilearn/datasets/func.py

```diff
@@ -1943,22 +2027,65 @@ def fetch_openneuro_dataset(
     downloaded_files : list of string
         Absolute paths of downloaded files on disk.
 
-    """
-    data_prefix = '{}/{}/uncompressed'.format(
-        dataset_version.split('_')[0], dataset_version)
-    data_dir = _get_dataset_dir(data_prefix, data_dir=data_dir,
-                                verbose=verbose)
+    Notes
+    -----
+    The default dataset downloaded by this function is the
+    "UCLA Consortium for Neuropsychiatric Phenomics LA5c" dataset
+    :footcite:p:`poldrack_phenome-wide_2016`.
+
+    This copy includes filenames that are not compliant with the current
+    version of :term:`BIDS`, so this function also calls
+    :func:`patch_openneuro_dataset` to generate BIDS-compliant symlinks.
+
+    See Also
+    --------
+    :func:`fetch_ds000030_urls`
+    :func:`patch_openneuro_dataset`
 
+    References
+    ----------
+    .. footbibliography::
+    """
     # if urls are not specified we download the complete dataset index
     if urls is None:
-        _, urls = fetch_openneuro_dataset_index(
-            data_dir=data_dir, dataset_version=dataset_version,
-            verbose=verbose)
+        DATASET_VERSION = 'ds000030_R1.0.4'
+        if DATASET_VERSION != dataset_version:
+            warnings.warn(
+                'If `dataset_version` is not "ds000030_R1.0.4", '
+                '`urls` must be specified. Downloading "ds000030_R1.0.4".'
+            )
+
+        data_prefix = '{}/{}/uncompressed'.format(
+            DATASET_VERSION.split('_')[0],
+            DATASET_VERSION,
+        )
+        data_dir = _get_dataset_dir(
+            data_prefix,
+            data_dir=data_dir,
+            verbose=verbose,
+        )
+
+        _, urls = fetch_ds000030_urls(
+            data_dir=data_dir,
+            dataset_version=DATASET_VERSION,
+            verbose=verbose,
+        )
+    else:
+        data_prefix = '{}/{}/uncompressed'.format(
+            dataset_version.split('_')[0],
+            dataset_version,
+        )
+        data_dir = _get_dataset_dir(
+            data_prefix,
+            data_dir=data_dir,
+            verbose=verbose,
+        )
 
     # The files_spec needed for _fetch_files
     files_spec = []
     files_dir = []
     for url in urls:
+        # NOTE: This seems fragile. What if data_prefix isn't in the URL?
```

Comment:
I agree (esp. when this is provided by the user...), I would at least add an explicit check before trying to split, and raise a comprehensive error if this is not the case.

Reply:
Done!